### PR TITLE
[Enhancement] Import starcache submodule to support DLA block cache.

### PR DESCRIPTION
**Pull request by GavinMar**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Import `starcache` submodule as a new block cache engine which support KV interfaces, and provides memory and disk cache. Currently, we support both `cachelib` and `starcache` engines, and we set the `starcache` as the default engine.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "starcache"]
+	path = starcache
+	url = https://github.com/StarRocks/starcache.git
```
Comment (imay) [inline .gitmodules:3]: Need to depend on a specified version?

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "starcache"]
+	path = starcache
```
Comment (DorianZheng) [inline .gitmodules:2]: better to put it under `be/src/thirdparty/starcache`. if we have multiple submodules in the future, it's more managable

File: build.sh [hunk 3]
```diff
@@ -267,6 +266,37 @@ else
     export LIBRARY_PATH=${JAVA_HOME}/jre/lib/amd64/server/
 fi
 
+build_submodules()
```
Comment (DorianZheng) [inline build.sh:269]: better to extract them into `common.sh`, it's duplicated in `run-be-ut.sh` too.

**Review by DorianZheng — CHANGES_REQUESTED** [review]

**Comment by DorianZheng** [comment]
left comments, rest LGTM.

File: .gitmodules (hunk 1 above)
Comment (imay) [reply .gitmodules:3]: Good to know, how to achieve it?

Comment (GavinMar) [reply .gitmodules:3]: > Good to know, how to achieve it?

Every time we manually pull the starcache submodule, the commit id will be updated to files 
 in starrocks/.git/module/starcache directory, such as the `HEAD` file.

File: .gitmodules (hunk 2 above)
Comment (GavinMar) [reply .gitmodules:2]: > better to put it under `be/src/thirdparty/starcache`. if we have multiple submodules in the future, it's more managable

OK

File: build.sh (hunk 3 above)
Comment (GavinMar) [reply build.sh:269]: > better to extract them into `common.sh`, it's duplicated in `run-be-ut.sh` too.

OK

**Review by DorianZheng — DISMISSED** [review]

**Review by DorianZheng — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**